### PR TITLE
BREAKING CHANGES: major bump upgrade ember@4.4.3

### PR DESCRIPTION
**Pull request by yk634**
## Context

Update to LTS version (4.4) as the version of Ember framework used is out of date.

## Objective

Modify code, tests to adapt to newer versions of Ember.

- Library Changes
  - ember-cli-jstree -> [ember-basic-tree](https://github.com/dwickern/ember-basic-tree)
    - Because ember-cli-jstree does not support Ember@4.
- High-impact updates to the library
  - `ember-bootstrap: "^5.0.0"`
    - bootstrap@3 deprecated in Ember@4.
- Newly started using library
  - [ember-svg-jar](https://github.com/voltidev/ember-svg-jar)
    - Used to display some icons.

We have confirmed that each function of Screwdriver.cd is available from the UI, except for the following
- UI does not follow when referring to Arifact from URL in Artifact view page.

## References

screwdriver-cd/screwdriver#2608

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: tests/integration/components/artifact-tree/component-test.js [hunk 1]
```diff
@@ -46,56 +46,43 @@ module('Integration | Component | artifact tree', function (hooks) {
...
   });
 
-  test('it renders with artifacts with artifact preselected', async function (assert) {
-    await render(hbs`
-      {{artifact-tree
-        buildStatus="SUCCESS"
-        selectedArtifact="coverage/coverage.json"
-      }}
-    `);
+  todo('it renders with artifacts with artifact preselected', async function (assert) {
```
Comment (ibu1224) [inline tests/integration/components/artifact-tree/component-test.js:69]: This feature now fails due to a change in the library used for arifact-previw
We will implement the feature later.

**Comment by ibu1224** [comment]
@screwdriver-cd/admin-all 
Ember.js upgrade is done. Please review.
The commit is a bit complicated, but please sqush merge when merging!

File: app/application/controller.js [hunk 2]
```diff
@@ -1,9 +1,10 @@
-import { computed } from '@ember/object';
+// import { computed } from '@ember/object';
```
Comment (tkyi) [inline app/application/controller.js:1]:
```suggestion
```

File: app/commands/detail/controller.js [hunk 3]
```diff
@@ -21,35 +22,42 @@ export default Controller.extend({
...
+      get() {
+        const version = this.selectedVersion || this.get('latest.version');
 
-      let { commandTagData } = this.commands;
+        const { versionOrTagFromUrl } = this.commands;
 
-      if (versionOrTagFromUrl === undefined) {
-        return this.commands.commandData.findBy('version', version);
-      }
+        const { commandTagData } = this.commands;
```
Comment (tkyi) [inline app/commands/detail/controller.js:40]: const { versionOrTagFromUrl, commandTagData } = this.commands;

File: app/components/app-header/styles.scss [hunk 4]
```diff
@@ -97,55 +102,37 @@
...
-
-    a {
-      color: $sd-white;
-      text-decoration: none;
-      display: inline-block;
-
       .img {
         vertical-align: middle;
-        fill: $sd-white;
+        // fill: $sd-white;
```
Comment (tkyi) [inline app/components/app-header/styles.scss:131]:
```suggestion
```

File: app/components/build-banner/component.js [hunk 5]
```diff
@@ -25,9 +25,17 @@ export default Component.extend({
 
   prNumber: computed('event.pr.url', {
     get() {
-      let url = this.getWithDefault('event.pr.url', '');
+      if (this._prNumber) {
+        return this._prNumber;
+      }
```
Comment (tkyi) [inline app/components/build-banner/component.js:30]: 👍

File: app/components/build-log/template.hbs [hunk 6]
```diff
@@ -1,32 +1,53 @@
...
+              Downloading
+            </span>
           {{else}}
-            <a onClick={{action "download"}} title="Download">{{fa-icon "download"}}</a>
+            <a onClick={{action "download"}} title="Download">
+              <FaIcon @icon="download" />
+            </a>
           {{/if}}
-          <a id="downloadLink" href download hidden></a>
+          <a id="downloadLink" href="" download="" hidden=""></a>
```
Comment (tkyi) [inline app/components/build-log/template.hbs:50]: is this right?

File: app/dashboard/show/controller.js [hunk 7]
```diff
@@ -1,8 +1,11 @@
+import { set } from '@ember/object';
 import { alias } from '@ember/object/computed';
-import { inject as service } from '@ember/service';
 import Controller from '@ember/controller';
+import { inject as service } from '@ember/service';
 
 export default Controller.extend({
+  store: service(),
```
Comment (tkyi) [inline app/dashboard/show/controller.js:7]: is this needed?

File: app/event/model.js [hunk 8]
```diff
@@ -183,8 +184,8 @@ export default DS.Model.extend(ModelReloaderMixin, {
 
         // See if any builds are running
         const runningBuild = list.find(b => {
-          const status = get(b, 'status');
-          const endTime = get(b, 'endTime');
+          const { status } = b;
+          const { endTime } = b;
```
Comment (tkyi) [inline app/event/model.js:188]:
```suggestion
          const { status, endTime } = b;
```

File: app/styles/app.scss [hunk 9]
```diff
@@ -56,7 +56,7 @@ $weight-bold: 600;
   box-shadow: none;
 }
 
-@import 'font-awesome';
+// @import 'font-awesome';
```
Comment (tkyi) [inline app/styles/app.scss:59]:
```suggestion
```

File: app/templates/detail/controller.js [hunk 10]
```diff
@@ -26,36 +27,44 @@ export default Controller.extend({
...
+    'templates.templateData.[]',
+    {
+      get() {
+        const version = this.selectedVersion || this.get('latest.version');
 
-      let { versionOrTagFromUrl } = this.templates;
+        const { versionOrTagFromUrl } = this.templates;
 
-      let { templateTagData } = this.templates;
+        const { templateTagData } = this.templates;
```
Comment (tkyi) [inline app/templates/detail/controller.js:40]: const { versionOrTagFromUrl, templateTagData } = this.templates;

File: config/environment.js [hunk 11]
```diff
@@ -1,11 +1,12 @@
 'use strict';
 
 module.exports = environment => {
-  let ENV = {
+  const ENV = {
     modulePrefix: 'screwdriver-ui',
     environment,
     rootURL: '/',
     locationType: 'auto',
+    // locationType: 'history',
```
Comment (tkyi) [inline config/environment.js:9]:
```suggestion
```

Comment (adong) [reply config/environment.js:9]: Actually, this one I think it needs to revert to `history`, see https://guides.emberjs.com/v4.4.0/configuring-ember/specifying-url-type/

File: app/dashboard/show/controller.js (hunk 7 above)
Comment (ibu1224) [reply app/dashboard/show/controller.js:6]: This seems necessary. Removing it caused several tests to fail.

**Comment by ibu1224** [comment]
@screwdriver-cd/admin-all 
Hi, @SDAdmins.
We are here to discuss the PR for the screwdriver.cd UI.

If you are developing with an older version of Ember.js, the code needs to be changed to a state supported by this PR.

Are you planning to add any new features ?
If there are no, I think it will be easier to proceed with development if you upgrade to Ember 4 first.

**Comment by adong** [comment]
> @screwdriver-cd/admin-all Hi, @SDAdmins. We are here to discuss the PR for the screwdriver.cd UI.
> 
> If you are developing with an older version of Ember.js, the code needs to be changed to a state supported by this PR.
> 
> Are you planning to add any new features ? If there are no, I think it will be easier to proceed with development if you upgrade to Ember 4 first.

I just merged one PR https://github.com/screwdriver-cd/ui/pull/880, and we can pause other PRs, and merge this PR first.

**Review by tkyi — DISMISSED** [review]

**Review by tkyi — APPROVED** [review]

**Comment by adong** [comment]
This is epic, lets merge it.

**Comment by jithine** [comment]
Nice work team 👏  Let's test this in open source SD https://cd.screwdriver.cd/

Some quick things which we noticed, mostly around CSS
1. font looks a bit large
2. CSS for tables in Collections & Commands looks odd
3. Access tokens section CSS in Secrets page

**Comment by yk634** [comment]
@jithine 
We are now fixing the problems caused by the Ember upgrade.

> CSS for tables in Collections & Commands looks odd

Can you tell us specifically what the problem is?
The following points are being recognized and corrected as problems, are there others?
- Sorting by Name, Namespace does not work in the table on the commands page.
- The position of the Name value on the collection page is skewed upward.

**Comment by adong** [comment]
I added this issue https://github.com/screwdriver-cd/screwdriver/issues/2840, that artifacts no longer working properly.

**Comment by jithine** [comment]
@yk634 sorry I meant Templates & Commands, Collections looks okay. 
For Templates & Commands the table CSS looks like plain table. Quite different that what we had before

**Pull request merged**